prow/genjobs/cmd/genjobs/main.go

```diff
@@ -189,31 +183,61 @@ func convertOrgRepoStr(o options, s string) string {
 	return strings.Join([]string{o.orgMap[org], repo}, "/")
 }
 
-// updateUtilityConfig updates the jobs UtilityConfig fields based on provided inputs to work with private repositories.
+// updateUtilityConfig updates the jobs UtilityConfig fields based on provided inputs.
 func updateUtilityConfig(o options, job *config.UtilityConfig) {
+	if o.bucket == "" && o.sshKeySecret == "" {
+		return
+	}
+
 	if job.DecorationConfig == nil {
-		job.DecorationConfig = &prowjob.DecorationConfig{
-			GCSConfiguration: &prowjob.GCSConfiguration{
-				Bucket: o.bucket,
-			},
-			SSHKeySecrets: []string{o.sshKeySecret},
+		job.DecorationConfig = &prowjob.DecorationConfig{}
+	}
+
+	updateGCSConfiguration(o, job.DecorationConfig)
+	updateSSHKeySecrets(o, job.DecorationConfig)
+}
+
+// updateGCSConfiguration updates the jobs GCSConfiguration fields based on provided inputs.
+func updateGCSConfiguration(o options, job *prowjob.DecorationConfig) {
+	if o.bucket == "" {
+		return
+	}
+
+	if job.GCSConfiguration == nil {
+		job.GCSConfiguration = &prowjob.GCSConfiguration{
+			Bucket: o.bucket,
 		}
 	} else {
-		if job.DecorationConfig.GCSConfiguration == nil {
-			job.DecorationConfig.GCSConfiguration = &prowjob.GCSConfiguration{
-				Bucket: o.bucket,
-			}
-		} else {
-			job.DecorationConfig.GCSConfiguration.Bucket = o.bucket
-		}
-		if job.DecorationConfig.SSHKeySecrets == nil {
-			job.DecorationConfig.SSHKeySecrets = []string{o.sshKeySecret}
-		} else {
-			job.DecorationConfig.SSHKeySecrets = append(job.DecorationConfig.SSHKeySecrets, o.sshKeySecret)
-		}
+		job.GCSConfiguration.Bucket = o.bucket
+	}
+}
+
+// updateSSHKeySecrets updates the jobs SSHKeySecrets fields based on provided inputs.
+func updateSSHKeySecrets(o options, job *prowjob.DecorationConfig) {
+	if o.sshKeySecret == "" {
+		return
+	}
+
+	if job.SSHKeySecrets == nil {
+		job.SSHKeySecrets = []string{o.sshKeySecret}
+	} else {
+		job.SSHKeySecrets = append(job.SSHKeySecrets, o.sshKeySecret)
 	}
 }
 
+// updateReporterConfig updates the jobs ReporterConfig fields based on provided inputs.
+func updateReporterConfig(o options, job *config.JobBase) {
+	if o.channel == "" {
+		return
+	}
+
+	if job.ReporterConfig == nil {
+		job.ReporterConfig = &prowjob.ReporterConfig{}
+	}
+
+	job.ReporterConfig.Slack = &prowjob.SlackReporterConfig{Channel: o.channel}
```

Comment:
It seems like it would be better to have prow handle this transparently. AKA add a reportSlack field

Reply:
Do you mean something like this https://github.com/kubernetes/test-infra/pull/15159? This flag is only *temporary* until this is supported in Prow config.

Reply:
Fabulous